packages/astro/src/core/build/generate.ts

```diff
@@ -125,7 +125,73 @@ export async function generatePages(options: StaticBuildOptions, internals: Buil
 
 		const assetsTimer = performance.now();
 		for (const [originalPath, transforms] of staticImageList) {
-			await generateImagesForPath(originalPath, transforms, assetsCreationPipeline, queue);
+			// Process each source image in parallel based on the queue’s concurrency
```

Comment:
I'm going to remove some of this and put it in this PR. Much as I'm a fan of lots of comments, it doesn't really need the full discussion of options in here - just a note on what's happening and a warning to read the PR before changing it